### PR TITLE
[RW-6352][risk=no] attempt at de-angularizing cdrVersionStore

### PR DESCRIPTION
**Pull request by als364**
So something hilarious happened here. We tried to make an atom store instead of a `ReplaySubject` store for cdrVersionStore, updated all the references from the `ReplaceSubject` store to the atom store, found that `homepage.spec.tsx` was still failing because it wasn't ever setting the `cdrVersionStore`, fixed that, THEN found that it was failing because our stubs for `getUserRecentResources` and `getWorkspaces` were returning disjoint workspaces.

We then spent an hour and a half trying to figure out why the test was passing without our changes, and eventually discovered this comment on `withCdrVersions`:
```
// HOC that provides a 'cdrVersionListResponse' prop with the CDR version
// information. Rendering of the connected component is blocked on initial load
// of the CDR versions. This should only affect initial page loads, this HOC can
// be included last (if multiple HOCs are in use) to minimize this impact.
```

The `HomepageComponent` was blocking rendering on a value that was never getting set. When we set the `cdrVersionStore` the tests failed as expected.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/homepage/recent-resources.tsx [hunk 1]
```diff
@@ -77,7 +76,7 @@ interface Props {
   workspaces: WorkspaceResponse[];
 }
 
-const RecentResources = fp.flow(withCdrVersions())((props: Props) => {
```
Comment (jmthibault79) [inline ui/src/app/pages/homepage/recent-resources.tsx:80]: this component needs `cdrVersionListResponse` in the Props - does it get set elsewhere?

Comment (als364) [reply ui/src/app/pages/homepage/recent-resources.tsx:80]: nope I just forgot to put it back

**Review by dmohs — APPROVED** [review]
Net 33 lines removed for the win!
👍 👍 👍

File: ui/src/app/utils/index.tsx [hunk 2]
```diff
@@ -400,7 +400,7 @@ export const withRouteConfigData = () => {
 // of the CDR versions. This should only affect initial page loads, this HOC can
 // be included last (if multiple HOCs are in use) to minimize this impact.
```
Comment (calbach) [inline ui/src/app/utils/index.tsx:401]: While this comment was true with `connectReplaySubject`, I don't think it's true anymore for `withStore` or the new subscribable, which appears to immediately render regardless of whether or not there's a value. Also, with the new store, there's a default "empty" value seeded into the store.

I suspect this causes a race condition on initial page render which can cause the initial page load to fail, e.g.:
https://app.circleci.com/pipelines/github/all-of-us/workbench/8720/workflows/2f20c5b4-98f7-4996-897d-3b3f7bda6a09/jobs/128585/artifacts

Possible resolutions:
- Make a different HOC which blocks child rendering until a store value is present, and remove the initial stub response from the store
- Refactor all of the clients of `withCdrVersions` to instead handle null/empty CDR versions (again, I think I would remove the stub response and use something like `null` or `undefined` to indicate that it's still loading).
- Move seeding of the cdr versions store somewhere early in the bootstrapping process, and block the main rendering path on it, e.g. in the signed in component or app router.

**Pull request merged**